### PR TITLE
Tracing: Add regex support for span filters

### PR DESCRIPTION
**Pull request by ektasorathia**
**What is this feature?**

Allows user to perform regex for span filter by tags

**Why do we need this feature?**

This feature will allow filtering the spans by tags performing a regex operation

**Who is this feature for?**

Operators and developers

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/TraceView/components/utils/filter-spans.tsx [hunk 1]
```diff
@@ -109,25 +109,16 @@ const getTagMatches = (spans: TraceSpan[], tags: Tag[]) => {
...
-            (span.instrumentationLibraryVersion &&
-              tag.key === LIBRARY_VERSION &&
-              tag.value === span.instrumentationLibraryVersion) ||
-            (span.traceState && tag.key === TRACE_STATE && tag.value === span.traceState) ||
-            (tag.key === ID && tag.value === span.spanID)
-          ) {
+          if (tag.operator=== '=' && checkKeyValConditionForMatch(tag, span)) {
             return getReturnValue(tag.operator, true);
+          } else if (tag.operator === '=~' && checkKeyValConditionForRegex(tag, span)) {
+            return getReturnValue(tag.operator, false);
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/components/utils/filter-spans.tsx:115]:
```suggestion
          } else if (tag.operator === '=~' && checkKeyValConditionForRegex(tag, span)) {
            return getReturnValue(tag.operator, true);
```

Will also have to modify `getReturnValue` right?

**Review by joey-grafana — CHANGES_REQUESTED** [review]

**Comment by joey-grafana** [comment]
Please also ensure all frontend tests pass and that you run the frontend linter.

File: public/app/features/explore/TraceView/components/utils/filter-spans.tsx (hunk 1 above)
Comment (ektasorathia) [reply public/app/features/explore/TraceView/components/utils/filter-spans.tsx:115]: updated the getReturnValue method but not sure if it's needed now. Earlier, the method had binary operation supporting = or != but now, it supports 4 operations. Can you please check if this method is required? Thanks

File: public/app/features/explore/TraceView/components/utils/filter-spans.test.ts [hunk 2]
```diff
@@ -334,6 +335,18 @@ describe('filterSpans', () => {
...
+        { ...defaultFilters, tags: [{ ...defaultTagFilter, key: 'tagKey1', operator: '=~', value: 'tagValue' }] },
+        spans
+      )
+    ).toEqual(new Set([spanID0, spanID2]));
+    expect(
+      filterSpans(
+        { ...defaultFilters, tags: [{ ...defaultTagFilter, key: 'tagKey1', operator: '!~', value: 'tagValue1' }] },
+        spans
+      )
+    ).toEqual(new Set([spanID2]));
```
[the remark this answers is not in the document]
Comment (ektasorathia) [reply public/app/features/explore/TraceView/components/utils/filter-spans.test.ts:355]: tagValue in the first test is a partial value. exact match is tagValue1 and tagValue0

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**